packages/shared/lib/contexts/onboarding/stores/shimmer-claiming-transactions.store.ts

```diff
@@ -0,0 +1,44 @@
+import { get, Writable } from 'svelte/store'
+
+import { InvalidTransactionIdError } from '@core/wallet'
+import { persistent } from '@lib/helpers'
+
+import { IShimmerClaimingTransactionStore } from '../interfaces'
+
+import { onboardingProfile } from './onboarding-profile.store'
+
+export const shimmerClaimingTransactions: Writable<IShimmerClaimingTransactionStore> = persistent(
+    'shimmerClaimingTransactions',
+    {}
+)
+
+export function isShimmerClaimingTransaction(transactionId: string, profileId?: string): boolean {
+    const _shimmerClaimingTransactions = get(shimmerClaimingTransactions)
+    profileId = profileId ? profileId : get(onboardingProfile)?.id
+    return profileId in _shimmerClaimingTransactions && _shimmerClaimingTransactions[profileId][transactionId]
+}
+
+export function persistShimmerClaimingTransaction(transactionId: string, profileId?: string): void {
+    if (!transactionId) {
+        throw new InvalidTransactionIdError()
```

Comment:
It isn't a invalid transactionId, but rather a missing one. Is this case ever gonna happen though?

Reply:
That's true. It's highly unlikely that this will happen, although if this error were to occur for any reason then it's likely we know we it came from. I can rename to `MissingTransactionIdError`